libctru/source/services/hid.c

```diff
@@ -42,6 +42,18 @@ Result hidInit(u32* sharedMem)
 
 	// Map HID shared memory at addr "sharedMem".
 	hidSharedMem=sharedMem;
+	if(!hidSharedMem)
```

Comment:
hidInit's sharedMem parameter should have been removed, instead of preserving it and managing ownership.